core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/ldap/ActiveDirectoryUserStoreManager.java

```diff
@@ -625,6 +628,142 @@ protected void doSetUserAttribute(String userName, String attributeName, String
+            searchCtls.setTimeLimit(searchTime);
+            searchCtls.setCountLimit(givenMax);
+
+            String groupSearchBase = realmConfig.getUserStoreProperty(LDAPConstants.GROUP_SEARCH_BASE);
+            String userListFilter = realmConfig.getUserStoreProperty(LDAPConstants.USER_NAME_LIST_FILTER);
+            String memberOFAttribute = realmConfig.getUserStoreProperty(LDAPConstants.MEMBEROF_ATTRIBUTE);
+            String groupNameAttribute = realmConfig.getUserStoreProperty(LDAPConstants.GROUP_NAME_ATTRIBUTE);
+            userSearchBase = realmConfig.getUserStoreProperty(LDAPConstants.USER_SEARCH_BASE);
+            String searchFilter = "(&" + userListFilter + "(" + memberOFAttribute + "=" + groupNameAttribute + "="
+                    + escapeSpecialCharactersForFilter(context.getRoleName()) + "," + groupSearchBase + "))";
+            String userNameProperty = realmConfig.getUserStoreProperty(LDAPConstants.USER_NAME_ATTRIBUTE);
+            String displayNameAttribute = realmConfig
+                    .getUserStoreProperty(LDAPConstants.DISPLAY_NAME_ATTRIBUTE);
+            String returnedAtts[] = {userNameProperty, displayNameAttribute};
+            searchCtls.setReturningAttributes(returnedAtts);
+
+            SearchResult sr = null;
+            dirContext = connectionSource.getContext();
+
+            answer = dirContext.search(escapeDNForSearch(userSearchBase), searchFilter, searchCtls);
+
+            for (int index = 0; index < givenMax && answer.hasMore(); index++) {
+                String displayName = null;
+                String userName = null;
+                sr = answer.next();
+                Attributes userAttributes = sr.getAttributes();
+                if (userAttributes != null) {
+                    Attribute userNameAttribute = userAttributes.get(userNameProperty);
+                    if (userNameAttribute != null) {
+                        userName = (String) userNameAttribute.get();
+                        if (debug) {
+                            logger.debug("UserName: " + userName);
+                        }
+                    }
+                    if (StringUtils.isNotEmpty(displayNameAttribute)) {
+                        Attribute displayAttribute = userAttributes.get(displayNameAttribute);
+                        if (displayAttribute != null) {
+                            displayName = (String) displayAttribute.get();
+                        }
+                        if (debug) {
+                            logger.debug("DisplayName: " + displayName);
+                        }
+                    }
+
+                    String domainName = realmConfig.getUserStoreProperty(
+                            UserCoreConstants.RealmConfig.PROPERTY_DOMAIN_NAME);
+
+                         /*
+                         Username will be null in the special case where the
+                         username attribute has changed to another
```

Comment:
Check the comment alignment.

Reply:
Resolved in https://github.com/wso2/carbon-kernel/pull/2748/commits/fac79311085b92b3378937bc4b20849e07868924